cocotb/runner.py

```diff
@@ -805,6 +804,112 @@ def test_command(self) -> List[Command]:
         return [[out_file] + self.plus_args]
 
 
+class Xcelium(Simulator):
+    @staticmethod
+    def simulator_in_path() -> None:
+        if shutil.which("xrun") is None:
+            raise SystemExit("ERROR: xrun executable not found!")
+
+    def check_toplevel_lang(self, toplevel_lang: Optional[str]) -> str:
+        if toplevel_lang is None:
+            if self.vhdl_sources and not self.verilog_sources:
+                return "vhdl"
+            elif self.verilog_sources and not self.vhdl_sources:
+                return "verilog"
+            else:
+                raise ValueError("Must specify a toplevel_lang in a mixed design")
+        elif toplevel_lang in ("verilog", "vhdl"):
+            return toplevel_lang
+        else:
+            raise ValueError(
+                f"Xcelium does not support {toplevel_lang!r} as a toplevel_lang"
+            )
```

Comment:
This seems to be repeating maybe can put this in the `Simulator` class?

Reply:
Indeed, I'll do this in another PR.